### PR TITLE
[youtube] Use the full Invidious site list in playlist, tab extractors

### PR DESCRIPTION
**Pull request by rhendric**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

For yt-dlp to handle Invidious links containing, e.g., playlist data, the other info extractors in youtube.py need to use the same list of Invidious sites that the main YoutubeIE class uses.

### CODE REVIEW

**Comment by pukkandan** [comment]
The list of instances are maintained for the video extractor only for code-similarity with youtube-dl. We can detect all invidious video pages automatically using the `link rel="alternate"` tag. Ideally, we would want a similar detection for the playlists/channels

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -3063,18 +3063,21 @@ class YoutubeTabIE(YoutubeBaseInfoExtractor):
...
                                 (?:playlist|watch)\?.*?\blist=
                             )|
-                            (?!(?:%s)\b)  # Direct URLs
+                            (?!(?:%(direct)s)\b)  # Direct URLs
                         )
                         (?P<id>[^/?\#&]+)
-                    ''' % YoutubeBaseInfoExtractor._RESERVED_NAMES
+                    ''' % {
+        'direct': YoutubeBaseInfoExtractor._RESERVED_NAMES,
+        'invidious': '|'.join(YoutubeIE._INVIDIOUS_SITES),
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3079]: move the list to the base extractor

File: yt_dlp/extractor/youtube.py [hunk 2]
```diff
@@ -3063,18 +3063,21 @@ class YoutubeTabIE(YoutubeBaseInfoExtractor):
...
                                 feed/|hashtag/|
                                 (?:playlist|watch)\?.*?\blist=
                             )|
-                            (?!(?:%s)\b)  # Direct URLs
+                            (?!(?:%(direct)s)\b)  # Direct URLs
                         )
                         (?P<id>[^/?\#&]+)
-                    ''' % YoutubeBaseInfoExtractor._RESERVED_NAMES
+                    ''' % {
+        'direct': YoutubeBaseInfoExtractor._RESERVED_NAMES,
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3078]: `reserved_names` or `reserved` is a better name

**Comment by rhendric** [comment]
> Ideally, we would want a similar detection for the playlists/channels

I wasn't sure how mature the Invidious-interfacing part of the generic extractor was; it doesn't work for me (something involving an HTTP 403 response). Reusing the site list so that the YouTube extractors would get selected seemed like the easiest fix, but would you prefer I focus on figuring out why the generic extractor doesn't work for me and fixing that instead?

**Comment by pukkandan** [comment]
I can merge this first anyway. You can look into generic seperately

**Pull request merged**